bots/npm-repo-test-invoke

```diff
@@ -0,0 +1,174 @@
+            }
+        }
+
+        return sink.Sink(host, identifier, status)
+
+def install(repo, revision, ref, name):
+    if ref:
+        subprocess.check_call([ "git", "fetch", "origin", ref ])
+        subprocess.check_output([ "git", "checkout", "origin/" + ref])
+
+    subprocess.check_output([ "npm", "install", "--save", "git+https://github.com/{0}#{1}".format(repo, revision)])
+
+    subprocess.check_output([ "git", "add", os.path.join(BASE, "package.json")])
+    subprocess.check_output(["git", "checkout", "--detach"])
+
+    # Commit the changes, if nothing to commit then there is nothing to test
+    try:
+        subprocess.check_output([ "git", "commit", "-m", "Testing {}".format(name)])
+    except subprocess.CalledProcessError:
+        return None
+
+    return subprocess.check_output([ "git", "rev-parse", "HEAD" ]).strip()
+
+def main():
+    parser = argparse.ArgumentParser(description='Run cockpit integration tests on a remote dependency')
+    parser.add_argument('-j', '--jobs', dest="jobs", type=int,
+            default=os.environ.get("TEST_JOBS", 1), help="Number of concurrent jobs")
+    parser.add_argument('--rebase', help="Rebase onto the specific branch before testing")
+    parser.add_argument('-o', "--offline", action='store_true',
+            help="Work offline, don''t fetch new data from origin for rebase")
+    parser.add_argument('--publish', dest='publish', default=os.environ.get("TEST_PUBLISH", ""),
+            action='store', help='Publish results centrally to a sink')
+    parser.add_argument('-v', '--verbose', action='store_true', help='Verbose output')
+    parser.add_argument('context', help="The context or type of integration tests to run")
+    parser.add_argument('ref', nargs='?', help="The Git remote ref to pull")
+    opts = parser.parse_args()
+
+    git_base = github.determine_github_base()
+    repo = os.environ.get("GITHUB_BASE", git_base)
+    name = os.environ.get("TEST_NAME", "{0}-tests".format(repo))
+    revision = os.environ.get("TEST_REVISION")
+
+    if not revision:
+        sys.stderr.write("Npm install failed: missing revision\n")
+        sys.exit(1)
+
+    active_sink = None
+    if opts.publish:
+        active_sink = start_publishing(opts.publish, name, revision, opts.context)
+
+    try:
+        head = install(repo, revision, opts.ref, name)
+    except subprocess.CalledProcessError:
+        traceback.print_exc()
+        return 1
+
+    if not head:
+        sys.stderr.write("Cockpit is already at this version: Nothing to test")
+        return 1
+
+    if active_sink:
+        active_sink.flush(status={
+            "state": "pending",
+            "context": opts.context,
+            "description": "{0} [{1}]".format(github.TESTING, HOSTNAME),
+            "target_url": ":link"
+        })
+
+
+    del os.environ["TEST_REVISION"]
+
+    os.environ["TEST_REVISION"] = head
+    os.environ["GITHUB_REVISION"] = revision
+    os.environ["GITHUB_KNOWN_ISSUE_BASE"] = git_base
```

Comment:
We have that. The problem is that when we set that variable to something that is not the main cockpit project, we still need to report known issues against the original repo.